### ISSUE TITLE
RIDB Side Stream API

### ISSUE THREAD

**Issue by nicaford** [OpenGlobe/myamerica-devsummit#19]
The Internet contains a wealth of information and social media related to our federal parks and recreation areas.  Much of this information is geo-located and is accessible and searchable via an API.  The RIDB Side Stream API would allow developers to pull supplemental information from a variety of sources using a single API and RIDB’s identifiers for recreation resources.  This would greatly simplify the effort required by a developer to access these supplemental resources when working with RIDB data.

Anticipated sources of supplemental information provided by the RIDB Side Stream API include:  photo sharing websites, Wikipedia, Google Maps, travel website ratings, social media content, and weather information.  The application calling the API would specify which sources of information should be included in the results.

The API would work through a combination of real time proximity searches and from an index stored in a database that would be created through searches performed on a daily basis.  Real time searches would be performed where timeliness is essential, such as social media content.  Other searches will occur behind the scenes nightly for each recreation area with their results captured in the database.

Consumers of the new API would be websites, web applications, and mobile applications which are already working from RIDB information.  The API would use REST protocols and would provide payloads in JSON or XML just like RIDB.

To prove the usefulness, we would like to build a simple mapping mashup app during the developer summit to show how the API could be used.  This proof of concept would also serve as a reference for other developers looking to use the RIDB Side Stream API.

**Comment by sethfitz** [comment]
Agreed!  See also #17 and/or talk to @mojodna and @danrademacher.  We'll have data from Flickr and possible Instagram available for PADUS areas that intersect with RIDB recreation area points.  A map of the data can be found here: https://stamen.cartodb.com/u/stamen-org/viz/204afede-d5b0-11e4-86f2-42f01d4f858a/public_map

Stamen did the photo + social media angle for California as part of parks.stamen.com and its successor, caliparks.org:

http://content.stamen.com/mapping_the_intersection_between_social_media_and_open_spaces_in_ca
http://content.stamen.com/caliparks

The code for the site is open source: https://github.com/stamen/caliparks.org  The social media harvesting components might be of more interest, in the [node-harvester](https://github.com/stamen/caliparks.org/tree/node-harvester) branch.
